src/transformers/modeling_utils.py

```diff
@@ -812,6 +812,62 @@ def enforce_repetition_penalty_(self, lprobs, batch_size, num_beams, prev_output
                 else:
                     lprobs[i, previous_token] /= repetition_penalty
 
+    def finalize_generation_logscores(
+        self,
+        next_token_logits,
+        input_ids,
+        batch_size,
+        num_beams,
+        repetition_penalty,
+        no_repeat_ngram_size,
+        bad_words_ids,
+        cur_len,
+        min_length,
+        max_length,
+        eos_token_id,
+        temperature,
+        do_sample,
+    ):
+        # repetition penalty (from CTRL paper https://arxiv.org/abs/1909.05858)
+        if repetition_penalty != 1.0:
+            self.enforce_repetition_penalty_(
+                next_token_logits, batch_size, num_beams, input_ids, repetition_penalty,
+            )
+
+        if temperature != 1.0:
```

Comment:
I guess this makes only sense for sampling or do you think it could / should also be used for greedy / beam search?